packages/server/src/restaurants/profile/domain/UpdatePartnerProfileHandler.ts

```diff
@@ -20,19 +21,27 @@ class UpdatePartnerProfileHandler implements Handler<UpdatePartnerProfileRequest
   constructor(@InjectModel(Restaurant.name) private restaurantModel: Model<RestaurantDocument>) {}
 
   async exec(req: UpdatePartnerProfileRequest): Promise<null | undefined> {
-    const result = await this.restaurantModel.findOneAndUpdate(
-      { _id: req.id },
-      {
-        name: req.name,
-        description: req.description,
-        cuisineType: req.cuisineType,
-        tags: req.tags,
-        bankAccountNumber: req.bankAccountNumber,
-        phoneNumber: req.phoneNumber,
-      },
-    );
+    const filter = { _id: req.id };
+    const update = {
+      name: req.name ? req.name : null,
+      description: req.description ? req.description : null,
+      cuisineType: req.cuisineType ? req.cuisineType : null,
+      tags: req.tags ? req.tags : null,
+      bankAccountNumber: req.bankAccountNumber ? req.bankAccountNumber : null,
+      phoneNumber: req.phoneNumber ? req.phoneNumber : null,
+    };
+
+    const result = await this.restaurantModel.findOneAndUpdate(filter, update, { new: true });
 
     if (result === null) return null;
+
+    const isNotNull = (val) => val != null;
+
+    if (_.values(result?.toObject()).every(isNotNull)) {
```

Comment:
```suggestion
    if (!_.values(result?.toObject()).some(_.isNull)) {
```
Jak weźmiemy [wbudowaną w lodash](https://lodash.com/docs/4.17.15#isNull) to nie trzeba definiować tej pomocniczej.